### DISCUSSION TITLE
How to configure SRTO_PAYLOADSIZE

### DISCUSSION THREAD

**Discussion by zzhu-igolgi** [Haivision/srt#2718] [Q&A] [answered]
Encountered an "invalid message api" error with sendmsg() while transmitting with RTP packets. Directly modifying the co.zExpPayloadSize value in live mode from SRT_LIVE_DEF_PLSIZE to SRT_LIVE_MAX_PLSIZE solve the issue. So I'm looking for an api solution to change the payload size configuration. 

Thank you.

**Comment by ethouris** [comment, marked as answer]
```
int desired_size = SRT_LIVE_MAX_PLSIZE;
srt_setsockflag(socket_id, SRTO_PAYLOADSIZE, &desired_size, sizeof desired_size);
```

**Reply by zzhu-igolgi** [reply]
I tried this to both input and output socket but still got the same "Incorrect use of Message API" error as before.

**Reply by zzhu-igolgi** [reply]
Got it solved. Thanks a lot.
